### PR TITLE
refactor: add DTOs

### PR DESCRIPTION
**Pull request by juantejer4**
# Refactor 4th challenge ⚡⚡⚡

_Resolve_: [865d40jqd](https://app.clickup.com/t/865d40jqd)

## Description ⭐

I've added DTOs

### CODE REVIEW

**Review by facundodartayete — APPROVED** [review]

File: app/Http/Requests/CreateAirlineRequest.php [hunk 1]
```diff
@@ -20,4 +21,13 @@ public function rules(): array
...
         ];
     }
+
+    public function toDto(): AirlineData
+    {
+        $name = $this->input('name');
+        $description = $this->input('description');
+        $cities = $this->input('cities');
+
+        return new AirlineData($name, $description, $cities);
```
Comment (facundodartayete) [inline app/Http/Requests/CreateAirlineRequest.php]: `$dto = new AirlineData();`
`$dto->name = $this->input('name');`

Comment (facundodartayete) [reply app/Http/Requests/CreateAirlineRequest.php]: `return new AirlineData(name: $name, description: $description, cities: $cities);`

Comment (facundodartayete) [reply app/Http/Requests/CreateAirlineRequest.php]: return new AirlineData(
 description: $description,
name: $name,
  cities: $cities
);

**Review by facundodartayete — APPROVED** [review]

File: app/Http/Controllers/Flight/UpdateFlightController.php [hunk 2]
```diff
@@ -2,15 +2,17 @@
...
 
 class UpdateFlightController
 {
-    public function __invoke(UpsertFlightRequest $request, Flight $flight): JsonResponse
+    public function __invoke(UpsertFlightRequest $request, Flight $flight, UpdateFlightAction $action): JsonResponse
     {
-        $flight->update($request->validated());
-        return response()->json($flight);
+        $flightData = $request->toDto();
+        $updated = $action->execute($flightData, $flight);
```
Comment (facundodartayete) [inline app/Http/Controllers/Flight/UpdateFlightController.php]: in these cases the model param goes before the data param

File: app/Http/Controllers/Flight/UpdateFlightController.php [hunk 3]
```diff
@@ -2,15 +2,17 @@
...
 
+use App\Http\Actions\UpdateFlightAction;
 use App\Http\Requests\UpsertFlightRequest;
 use App\Models\Flight;
 use Illuminate\Http\JsonResponse;
 
 class UpdateFlightController
 {
-    public function __invoke(UpsertFlightRequest $request, Flight $flight): JsonResponse
+    public function __invoke(UpsertFlightRequest $request, Flight $flight, UpdateFlightAction $action): JsonResponse
```
Comment (facundodartayete) [inline app/Http/Controllers/Flight/UpdateFlightController.php]: in many casas you may need more than one action, that is why we tend to name the param exactly as the class ` UpdateFlightAction $updateFlightAction`

Comment (facundodartayete) [reply app/Http/Controllers/Flight/UpdateFlightController.php]: plus it is more readable when you use it below

File: app/Http/Actions/GetFlightsAction.php
Comment (facundodartayete) [inline app/Http/Actions/GetFlightsAction.php:1]: just like with controllers, we separate actions into folders, as they grow too big too quickly

Comment (facundodartayete) [inline app/Http/Actions/GetFlightsAction.php:1]: also, actions and dtos don't go inside the http folder, as they are not part of that layer, but don't worry on that refactor for now, just remember for next time

**Review by facundodartayete — APPROVED** [review]

File: app/Http/Actions/GetFlightsAction.php
Comment (juantejer4) [reply app/Http/Actions/GetFlightsAction.php:1]: Yes Sir

**Pull request merged**